tests/integration/access_grant_funding/routes/test_reports.py

```diff
@@ -547,6 +453,355 @@ def test_decline_sign_off_get(
         )
 
 
+class TestConfirmReportSubmission:
+    @pytest.mark.parametrize(
+        "client_fixture, can_access",
+        (
+            ("authenticated_no_role_client", False),
+            ("authenticated_grant_recipient_member_client", False),
+            ("authenticated_grant_recipient_data_provider_client", True),
+            ("authenticated_grant_recipient_certifier_client", False),
+        ),
+    )
```

Comment:
One small thing I tried to do at some point, is define this parametrization as a fixture itself - so that the list of clients+expectations is in one place. Makes it easier to update all tests when we add a new client, which I think we did recently.

Unsure if this is definitely a good idea - each new client then adds 10s/100s of new tests, which maybe is overkill. But does ensure more reliable checking at least? One to momentarily think about here perhaps.